content/blog/migrate-kubedns-to-coredns-in-kubernetes.md

```diff
@@ -0,0 +1,117 @@
+That is, if you had configured `stubdomains`, `upstreamnameservers` and `federation` via the kube-dns configMap, it will now be translated automatically to the equivalent CoreDNS configmap during when choosing to install CoreDNS using `kubeadm upgrade`.
+
+`Stubdomain` and `upstreamnameserver` in kube-dns translates to the [`proxy`](https://coredns.io/plugins/proxy/) in CoreDNS.
+The `federation` in kube-dns has an equivalent [`federation`](https://coredns.io/plugins/federation/) in CoreDNS.
+
+
+Example ConfigMap of kube-dns.
+~~~text
+apiVersion: v1
+data:
+  federations: |
+    {"foo" : "foo.feddomain.com"}
+  stubDomains: |
+    {"abc.com" : ["1.2.3.4"], "my.cluster.local" : ["2.3.4.5"]}
+  upstreamNameservers: |
+    ["8.8.8.8", "8.8.4.4"]
+kind: ConfigMap
+metadata:
+  creationTimestamp: 2018-01-22T20:21:56Z
+  name: kube-dns
+  namespace: kube-system
+~~~
+
+CoreDNS Corefile after translation.
+
+~~~text
+   .:53 {
+        errors
+        health
+        kubernetes cluster.local  in-addr.arpa ip6.arpa {
+           upstream  8.8.8.8 8.8.4.4
+           pods insecure
+           fallthrough in-addr.arpa ip6.arpa
+        }
+       federation cluster.local {
+           foo foo.feddomain.com
+        }
+        prometheus :9153
+        proxy .  8.8.8.8 8.8.4.4
+        cache 30
+    }
+    abc.com:53 {
+        errors
+        cache 30
+        proxy . 1.2.3.4
+    }
+    my.cluster.local:53 {
+        errors
+        cache 30
+        proxy . 2.3.4.5
+    }
+~~~
+
+
+
+## Installing CoreDNS via Minikube.
+
+CoreDNS is available in the `addon manager` disabled by default.
+
+~~~text
+$ minikube addons list
+- kube-dns: enabled
+- registry: disabled
+- registry-creds: disabled
+- freshpod: disabled
+- addon-manager: enabled
+- dashboard: enabled
+- coredns: disabled
+- heapster: disabled
+- efk: disabled
+- ingress: disabled
+- default-storageclass: enabled
+- storage-provisioner: enabled
+~~~
+
+To enable CoreDNS, run the following command:
+> NOTE: Ensure to disable kube-dns after enabling CoreDNS.
+
+~~~
+$ minikube addons enable coredns
+coredns was successfully enabled
+~~~
+
+
+## CoreDNS in kube-up
+
+[Kube-up](https://kubernetes.io/docs/getting-started-guides/scratch/) is another way to start a Kubernetes cluster, now mostly used for deploying Kubernetes in GCE for end-to-end (e2e) testing purposes.
+CoreDNS can be installed as the default DNS service, you need to set the environment variable `CLUSTER_DNS_CORE_DNS` to `true`.
+
+## CoreDNS in Kops
+
+Currently, Kops is set to include [CoreDNS as an option](https://github.com/kubernetes/kops/pull/4041) to be installed instead of kube-dns in v1.10.
+
+## Installing CoreDNS via other methods
+
+For users keen to install CoreDNS in place of kube-dns but are not using `kubeadm`, `minikube`, `kube-up`, or `kops`, there is an existing document available [here](https://github.com/coredns/deployment/tree/master/kubernetes), which will help you to migrate from kube-dns to CoreDNS.
```

Comment:
s/but are/but who are/

Reply:
s/there is an existing document [here]/there are instructions in the [CoreDNS deployment repository]/